src/ALE/MOM_regridding.F90

```diff
@@ -751,25 +752,18 @@ subroutine build_sigma_grid( CS, G, GV, h, dzInterface )
   do i = G%isc-1,G%iec+1
     do j = G%jsc-1,G%jec+1
 
+      ! The rest of the model defines grids integrating up from the bottom
+      nominalDepth = G%bathyT(i,j)*GV%m_to_H
+
       ! Determine water column height
+      zOld(nz+1) =  -nominalDepth
       totalThickness = 0.0
-      do k = 1,nz
+      do k = nz,1,-1
         totalThickness = totalThickness + h(i,j,k)
```

Comment:
Heads up: Changing the direction of this loop changed answers because the order of summation matters. I'm flipping this loop back to recover answers even though it's marginally less efficient to do so. The rest of the commit passes our tests.